metrics/vaultmetrics.go

```diff
@@ -0,0 +1,37 @@
+package metrics
+
+import (
+	"github.com/ecadlabs/signatory/signatory"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var vaultSigningSummary = prometheus.NewSummaryVec(
+	prometheus.SummaryOpts{
+		Name: "vault_sign_request_durations",
+		Help: "Vaults signing requests latencies in seconds",
+	}, []string{"vault"})
+
+type metricVault struct {
+	vault signatory.Vault
+}
+
+func (v *metricVault) Contains(keyHash string) bool { return v.vault.Contains(keyHash) }
+func (v *metricVault) GetPublicKey(keyHash string) (signatory.PublicKey, error) {
+	return v.vault.GetPublicKey(keyHash)
+}
+func (v *metricVault) ListPublicKeys() ([]signatory.PublicKey, error) { return v.vault.ListPublicKeys() }
+func (v *metricVault) Sign(digest []byte, key string, alg string) ([]byte, error) {
+	timer := prometheus.NewTimer(prometheus.ObserverFunc(func(val float64) {
+		us := val * 1000000 // make microseconds
```

Comment:
Can you use the appropriate constant from the `time` package. See: https://golang.org/pkg/time/#pkg-constants